scalafmt-core/shared/src/main/scala/org/scalafmt/util/LiteralOps.scala

```diff
@@ -0,0 +1,70 @@
+package org.scalafmt.util
+
+import org.scalafmt.config.{Case, ScalafmtConfig}
+
+object LiteralOps {
+
+  /**
+    * Prints integer literals with specified case
+    *
+    * Divides literal into three parts:
+    *
+    * 0xFF123L
+    * 0x is a hex prefix
+    * FF123 is a body
+    * L is a long suffix
+    *
+    * literals.long applies to suffix and literals.hex applies to body
+    * */
+  def prettyPrintInteger(
+      str: String
+  )(implicit style: ScalafmtConfig): String = {
+    val sb = new StringBuilder()
+    val body =
+      if (str.startsWith("0x") || str.startsWith("0X")) {
+        sb.append("0x")
+        str.drop(2)
+      } else {
+        str
+      }
+
+    if (body.endsWith("L") || body.endsWith("l")) {
+      sb.append(style.literals.hex.process(body.dropRight(1)))
+      sb.append(style.literals.long.process(body.takeRight(1)))
+    } else {
+      sb.append(style.literals.hex.process(body))
+    }
+
+    sb.toString()
+  }
+
+  def prettyPrintFloat(str: String)(implicit style: ScalafmtConfig): String =
+    prettyPrintFloatingPoint(str, "F", "f", style.literals.float)
+
+  def prettyPrintDouble(str: String)(implicit style: ScalafmtConfig): String =
+    prettyPrintFloatingPoint(str, "D", "d", style.literals.double)
+
+  /**
+    * Prints floating point literals with specified case
+    *
+    * Divides literals into two parts:
+    *
+    * 1.0e-10f
+    * 1.0e-10 is a body with scientific notation
+    * f is a float/double suffix
+    *
+    * literals.scientific applies to body and literals.float/double applies to suffix
+    * */
+  private def prettyPrintFloatingPoint(
+      str: String,
+      suffixUpper: String,
+      suffixLower: String,
+      suffixCase: Case
+  )(implicit style: ScalafmtConfig): String =
+    if (str.endsWith(suffixUpper) || str.endsWith(suffixLower)) {
+      style.literals.scientific.process(str.dropRight(1)) +
```

Comment:
are we assuming that the suffix contains one character? in that case, can we pass a Char instead of a String?

Reply:
are we assuming that the suffix contains one character?suffix of floating point literals can be

Reply:
sure, let's go with String but shouldn't we either verify its length is 1, or just use dropRight(suffix.length)?